cmd/disk-cache-config.go

```diff
@@ -52,12 +55,46 @@ func (cfg *CacheConfig) UnmarshalJSON(data []byte) (err error) {
 
 // Parses given cacheDrivesEnv and returns a list of cache drives.
 func parseCacheDrives(drives []string) ([]string, error) {
+	if len(drives) == 0 {
+		return drives, nil
+	}
+	var endpoints []string
 	for _, d := range drives {
+		if ellipses.HasEllipses(d) {
+			s, err := parseCacheDrivePaths(d)
+			if err != nil {
+				return nil, err
+			}
+			endpoints = append(endpoints, s...)
+		} else {
+			endpoints = append(endpoints, d)
+		}
+	}
+
+	for _, d := range endpoints {
 		if !filepath.IsAbs(d) {
 			return nil, uiErrInvalidCacheDrivesValue(nil).Msg("cache dir should be absolute path: %s", d)
 		}
 	}
-	return drives, nil
+	return endpoints, nil
+}
+
+// Parses all arguments and returns a slice of drive paths following the ellipses pattern.
+func parseCacheDrivePaths(args ...string) (ep []string, err error) {
+	var argPatterns = make([]ellipses.ArgPattern, len(args))
+	for i, arg := range args {
+		patterns, perr := ellipses.FindEllipsesPatterns(arg)
+		if perr != nil {
+			return []string{}, uiErrInvalidCacheDrivesValue(nil).Msg(perr.Error())
+		}
+		argPatterns[i] = patterns
+	}
+	for _, argPattern := range argPatterns {
+		for _, lbls := range argPattern.Expand() {
+			ep = append(ep, strings.Join(lbls, ""))
+		}
+	}
+	return ep, nil
```

Comment:
Now this function can be re-written as:
```
// Parses all arguments and returns a slice of drive paths following the ellipses pattern.
func parseCacheDrivePaths(arg string) (ep []string, err error) {
        patterns, perr := ellipses.FindEllipsesPatterns(arg)
        if perr != nil {
                return []string{}, uiErrInvalidCacheDrivesValue(nil).Msg(perr.Error())
        }
        for _, lbls := range patterns.Expand() {
                ep = append(ep, strings.Join(lbls, ""))
        }
        return ep, nil
}
```

@harshavardhana can you check if the APIs of ellipses package are being used how it's intended to be used? it is not obvious.

Reply:
The function is correct because @krishnasrinivas looks like this function can take multiple arguments like `/home/drives{1...3};/home/otherdrives{1...3}`

Reply:
@krishnasrinivas , updated PR per your review suggestions